plugins/kubernetes/app/models/kubernetes/release_doc.rb

```diff
@@ -205,6 +206,48 @@ def disruption_budget_target(deployment)
       end
     end
 
+    def create_env_config_map
+      return unless deployment = raw_template.detect { |r| ["Deployment", "StatefulSet"].include? r[:kind] }
+      return if raw_template.any? do |r|
+        r[:kind] == "ConfigMap" && r.dig(:metadata, :annotations, :"samson/envConfigMap")
+      end
+      return unless deployment.dig(:metadata, :annotations, :"samson/env_from_config_map")
+
+      name = [
+        deployment.dig_fetch(:metadata, :name),
+        # We always inject the blue-green color to ensure we only have previous and current without
+        # leaving old configmaps around or creating a race during deploys.
+        kubernetes_release.blue_green_color,
+        "env"
+      ].join('-')
+
+      annotations = (deployment.dig(:metadata, :annotations) || {}).slice(
+        :"samson/override_project_label"
+      ).merge!(
+        "samson/updateTimestamp": Time.now.utc.iso8601,
+        "samson/envConfigMap": true
+      )
+
+      cm = {
+        apiVersion: "v1",
+        kind: "ConfigMap",
+        metadata: {
+          name: name,
+          labels: deployment.dig_fetch(:metadata, :labels).dup,
+          annotations: annotations
+        },
+        immutable: true,
+        data: static_env
+      }
+
+      if deployment[:metadata].key?(:namespace)
+        cm[:metadata][:namespace] = deployment.dig(:metadata, :namespace)
+      end
+
+      raw_template << cm
```

Comment:
prefer that to happen outside the method for visibility, and ideally not change raw_template but just the returned templates if possible